x/batching/keeper/endblock.go

```diff
@@ -79,59 +105,56 @@ func (k Keeper) ConstructBatch(ctx sdk.Context) (types.Batch, [][]byte, [][]byte
 	batchID := hash.Sum(nil)
 
 	return types.Batch{
-		BatchNumber:     newBatchNum,
-		BlockHeight:     ctx.BlockHeight(),
-		DataResultRoot:  hex.EncodeToString(dataRoot),
-		ValidatorRoot:   hex.EncodeToString(valRoot),
-		BatchId:         batchID,
-		ProvingMedatada: nil,
+		BatchNumber:           newBatchNum,
+		BlockHeight:           ctx.BlockHeight(),
+		CurrentDataResultRoot: hex.EncodeToString(dataRoot),
+		DataResultRoot:        hex.EncodeToString(superRoot),
+		ValidatorRoot:         valRootHex,
+		BatchId:               batchID,
+		ProvingMedatada:       nil,
 	}, dataEntries, valEntries, nil
 }
 
 // ConstructDataResultTree constructs a data result tree based on the
 // data results that have not been batched yet. It returns the tree's
-// entries (unhashed leaf contents) and hex-encoded root.
+// entries without the domain separators and the tree root.
 func (k Keeper) ConstructDataResultTree(ctx sdk.Context, newBatchNum uint64) ([][]byte, []byte, error) {
 	dataResults, err := k.GetDataResults(ctx, false)
 	if err != nil {
 		return nil, nil, err
 	}
 
 	entries := make([][]byte, len(dataResults))
+	treeEntries := make([][]byte, len(dataResults))
 	for i, res := range dataResults {
 		resHash, err := hex.DecodeString(res.Id)
```

Comment:
Maybe it's helpful to name this `resID`? I get that it's the hash of the result, but it might help to avoid confusion whether the result ID is hashed again or not.

Reply:
Oh you mean `resHash` -> `resID`, right? Agreed fixed